dojo/tools/burp_graphql/parser.py

```diff
@@ -0,0 +1,196 @@
+import logging
+import json
+import re
+
+
+from dojo.models import Endpoint, Finding
+
+logger = logging.getLogger(__name__)
+
+
+class BurpGraphQLParser(object):
+
+    def get_scan_types(self):
+        return ["Burp GraphQL API"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return scan_type  # no custom label for now
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Import Burp Enterprise Edition findings from the GraphQL API"
+
+    def get_findings(self, filename, test):
+        if filename:
+            tree = filename.read()
+            try:
+                data = json.loads(str(tree, 'utf-8'))
+            except:
+                data = json.loads(tree)
+
+            scan_data = data.get('Issues')
+
+            return self.create_findings(scan_data, test)
```

Comment:
There is few things wrong in this part.
1. If filename is `None` the code do nothing. It can't happen but it's weird to see this `if` statement.
2. use the simpler form `data = json.load(filename)`
3. you could add a check on the `Issues` attribute that launch a `ValueError`. This check could catch if a wrong report is pushed to this parser

Reply:
Updated in latest commit (4d3ae9f)